### ISSUE TITLE
Failed step "analyzeCommits" of plugin "[Function: ]"

### ISSUE THREAD

**Issue by codeandcats** [pmowrer/semantic-release-monorepo#76]
When trying to run `semantic-release -e semantic-release-monorepo --dry-run`

With no release config in `package.json`...

I run into the following error:

```
[23:25:21] [semantic-release] › ✖  Failed step "analyzeCommits" of plugin "[Function: ]"
[23:25:21] [semantic-release] › ✖  An error occurred while running semantic-release: TypeError: Expected `moduleId` to be of type `string`, got `object`
```

Here's the full output:

```
[23:25:14] [semantic-release] › ℹ  Running semantic-release version 15.13.30
[23:25:15] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/npm"
[23:25:15] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/github"
[23:25:15] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/npm"
[23:25:15] [semantic-release] › ✔  Loaded plugin "publish" from "@semantic-release/npm"
[23:25:15] [semantic-release] › ✔  Loaded plugin "publish" from "@semantic-release/github"
[23:25:15] [semantic-release] › ✔  Loaded plugin "success" from "@semantic-release/github"
[23:25:15] [semantic-release] › ✔  Loaded plugin "fail" from "@semantic-release/github"
[23:25:15] [semantic-release] › ⚠  Run automated release from branch master in dry-run mode
[23:25:19] [semantic-release] › ✔  Allowed to push to the Git repository
[23:25:19] [semantic-release] › ℹ  Start step "verifyConditions" of plugin "@semantic-release/npm"
[23:25:19] [semantic-release] › ✔  Completed step "verifyConditions" of plugin "@semantic-release/npm"
[23:25:19] [semantic-release] › ℹ  Start step "verifyConditions" of plugin "@semantic-release/github"
[23:25:19] [semantic-release] [@semantic-release/github] › ℹ  Verify GitHub authentication
[23:25:19] [semantic-release] › ✔  Completed step "verifyConditions" of plugin "@semantic-release/github"
[23:25:21] [semantic-release] › ℹ  No git tag version found
[23:25:21] [semantic-release] › ℹ  No previous release found, retrieving all commits
[23:25:21] [semantic-release] › ℹ  Found 9 commits since last release
[23:25:21] [semantic-release] › ℹ  Start step "analyzeCommits" of plugin "[Function: ]"
[23:25:21] [semantic-release] › ✖  Failed step "analyzeCommits" of plugin "[Function: ]"
[23:25:21] [semantic-release] › ✖  An error occurred while running semantic-release: TypeError: Expected `moduleId` to be of type `string`, got `object`
    at resolveFrom (/<path-to-project>/node_modules/resolve-from/index.js:11:9)
    at module.exports (/<path-to-project>/node_modules/resolve-from/index.js:34:41)
    at module.exports (/<path-to-project>/node_modules/semantic-release-plugin-decorators/node_modules/import-from/index.js:4:49)
    at requirePlugin (/<path-to-project>/node_modules/semantic-release-plugin-decorators/src/index.js:4:33)
    at resolvePluginFn (/<path-to-project>/node_modules/semantic-release-plugin-decorators/src/index.js:18:18)
    at /<path-to-project>/node_modules/semantic-release-plugin-decorators/src/index.js:26:20
    at validator (/<path-to-project>/node_modules/semantic-release/lib/plugins/normalize.js:34:30)
    at /<path-to-project>/node_modules/semantic-release/lib/plugins/pipeline.js:37:40
    at next (/<path-to-project>/node_modules/p-reduce/index.js:17:9)
    at processTicksAndRejections (internal/process/task_queues.js:93:5) {
  pluginName: '[Function: ]'
}
TypeError: Expected `moduleId` to be of type `string`, got `object`
    at resolveFrom (/<path-to-project>/node_modules/resolve-from/index.js:11:9)
    at module.exports (/<path-to-project>/node_modules/resolve-from/index.js:34:41)
    at module.exports (/<path-to-project>/node_modules/semantic-release-plugin-decorators/node_modules/import-from/index.js:4:49)
    at requirePlugin (/<path-to-project>/node_modules/semantic-release-plugin-decorators/src/index.js:4:33)
    at resolvePluginFn (/<path-to-project>/node_modules/semantic-release-plugin-decorators/src/index.js:18:18)
    at /<path-to-project>/node_modules/semantic-release-plugin-decorators/src/index.js:26:20
    at validator (/<path-to-project>/node_modules/semantic-release/lib/plugins/normalize.js:34:30)
    at /<path-to-project>/node_modules/semantic-release/lib/plugins/pipeline.js:37:40
    at next (/<path-to-project>/node_modules/p-reduce/index.js:17:9)
    at processTicksAndRejections (internal/process/task_queues.js:93:5) {
  pluginName: '[Function: ]'
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed.
Exit code: 1
```

**Comment by JordanForeman** [comment]
possible duplicate of #65

I'm using the same version of `semantic-release` and seeing the same issue. Going to downgrade and try that out.

**Comment by gmlnchv** [comment]
@JordanForeman have you had any luck downgrading?

**Comment by hweeks** [comment]
@gmlnchv @JordanForeman @codeandcats Hey y'all, I've got an open PR to fix this issue. I've mentioned @pmowrer on it, but additional eyes may help get it merged in. 🎉

**Comment by folke** [comment]
Since I couldn't get this package to work properly, I've created a new package [semantic-release-commit-filter](https://www.npmjs.com/package/semantic-release-commit-filter) that provides pretty much the same functionality, but in only about **20 lines** of code 😃

Configuration of `semantic-release` is the same as in a non-monorepo.

Let me know what you think!

**Comment by pmowrer** [comment]
> It works by hijacking calls to the git-log-parser parse method.

Way to get creative... Love the solution! 

Once a couple of other issues were resolved by feature additions in `semantic-release`, commit filtering is about the only thing that's needed anymore.

**Comment by pmowrer** [comment]
I posted a branch with what I believe will fix the issue here: https://github.com/pmowrer/semantic-release-monorepo/pull/80. Let know if it works for you

**Comment by pmowrer** [comment]
This should be fixed by in [v7.0.0](https://github.com/pmowrer/semantic-release-monorepo/releases/tag/v7.0.0).

**Comment by pedropmota** [comment]
The new version works for me! Just started using the tool and loving it. Thanks!
